cmd/storage-rest-client.go

```diff
@@ -345,6 +353,21 @@ func (client *storageRESTClient) RenameFile(srcVolume, srcPath, dstVolume, dstPa
 	return err
 }
 
+// Gets peer storage server's instanceID - to be used with every REST call for validation.
+func (client *storageRESTClient) getInstanceID() (err error) {
+	respBody, err := client.restClient.Call(storageRESTMethodGetInstanceID, nil, nil, -1)
+	if err != nil {
+		return err
+	}
+	instanceIDBuf := make([]byte, 64)
+	n, err := io.ReadFull(respBody, instanceIDBuf)
```

Comment:
Yes thought about that, also thought about making client request with instanceID in the request header. But the problem is storage-rest-client does not have access to request/response headers. It will complicate the code for other StorageAPI calls as `rest.Client.Call()` API will have to expose the request/response.

Reply:
Okay then we can leave it as is, but we need to drain the respBody here properly.